cmd/analyze/main.go

```diff
@@ -167,7 +168,7 @@ func staticAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, resultStores *work
 	}
 }
 
-func main() {
+func run() (int, error) {
```

Comment:
Is there a compelling reason to distinguish the usage errors vs other failures? The code gets a lot simpler if you just return either status 0 or 1.

Reply:
Also, you could just return `error` and leave the exit code decision up to `main()`

Reply:
I was just thinking of matching other unix utilities in this way. Since the options are so complex, I think it helps for usability to distinguish user errors from program errors.

Reply:
Is anything checking for the difference though?
If I saw "exit code 1" or "exit code 2" I'd still need to look at the other logs to figure out what went wrong.
I think unless another program is going to interpret your exit code in some meaningful way, it's better to err on the side of simplicity and just return 0 or 1.

Reply:
Fair enough. I'll implement the error type as you suggested in the other comment but just return exit code 1 on any error until there's a clear need to return distinct error codes.